api/app/signals/apps/signals/utils/location.py

```diff
@@ -36,7 +36,79 @@ def _get_stadsdeel_code(geometry: PointField, default: Optional[str] = None) ->
     if not settings.FEATURE_FLAGS.get('API_DETERMINE_STADSDEEL_ENABLED', False):
         return default
 
+    from signals.apps.signals.models.location import AREA_STADSDEEL_TRANSLATION
+
     area_type = getattr(settings, 'API_DETERMINE_STADSDEEL_ENABLED_AREA_TYPE', 'sia-stadsdeel')
     area = _get_area(geometry=geometry, area_type=area_type)
     code = AREA_STADSDEEL_TRANSLATION.get(area.code.lower(), None) if area else None
     return code or default
+
+
+class AddressFormatter:
+    re_formatchars = re.compile(r'(?<!\\)([OhltTpPW])')
+    re_escaped = re.compile(r'\\(.)')
+
+    def __init__(self, address):
+        self.address = address
+
+    def O(self):  # noqa E743
+        """
+        Openbare ruimte
+        """
+        return self.address['openbare_ruimte'] if 'openbare_ruimte' in self.address else ''
+
+    def h(self):
+        """
+        Huisnummer
+        """
+        return self.address['huisnummer'] if 'huisnummer' in self.address else ''
+
+    def l(self):  # noqa E743
+        """
+        Huisletter
+        """
+        return self.address['huisletter'] if 'huisletter' in self.address else ''
+
+    def t(self):
+        """
+        Huisnummer toevoeging  without a hyphen
+        """
+        return self.address['huisnummer_toevoeging'] if 'huisnummer_toevoeging' in self.address else ''
+
+    def T(self):  # noqa E743
+        """
+        Huisnummer toevoeging with a hyphen
+        """
+        if 'huisnummer_toevoeging' in self.address and len(self.address['huisnummer_toevoeging'].strip()) > 0:
+            return f'-{self.address["huisnummer_toevoeging"]}'
```

Comment:
Added it so that in case of using this in the email template it will not result in multiple format declarations to get a hyphen there. If you added it the same way as time does the colon it can sometimes result in only having a hyphen, this will be the case if there is no `huisletter` present